### PR TITLE
Implemented tags API methods

### LINKED ISSUE #199: Feature requests: add tags support
Please correct me if I am wrong, but is the tag part of the gitlab API not currently supported in this client?

https://github.com/gitlabhq/gitlabhq/blob/master/doc/api/tag.md

If that's the case, I'd be happy to do the work to put them in.

### PR DESCRIPTION
**Pull request by jblackman**
This resolves #199.

Implemented tags API methods. Moved the existing tags() and create_tag() methods from repository.rb to tags.rb.

Sorry about the long delay between raising the FR and implementing it.

### CODE REVIEW

File: lib/gitlab/client/tags.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    #
+    # @example
+    #   Gitlab.tag(3, 'api')
+    #   Gitlab.repo_tag(5, 'master')
+    #
+    # @param  [Integer] project The ID of a project.
+    # @param  [String] tag The name of the tag.
+    # @return [Gitlab::ObjectifiedHash]
+    def tag(project, tag)
+      get("/projects/#{project}/repository/tags/#{tag}", query: {})
```
Comment (asedge) [inline lib/gitlab/client/tags.rb]: Can't we just leave `query: {}` off here?

**Comment by asedge** [comment]
Looks pretty good, just one comment that could use addressing.  Thanks!

File: lib/gitlab/client/tags.rb (hunk 1 above)
Comment (jblackman) [inline lib/gitlab/client/tags.rb]: Shall do

**Comment by asedge** [comment]
Thank you!

**Pull request merged**